python/pyarrow/io.pxi

```diff
@@ -1459,23 +1484,21 @@ cdef class BufferReader(NativeFile):
 
     Examples
     --------
+    Create an Arrow input stream and inspect it:
 
     >>> import pyarrow as pa
     >>> data = b'reader data'
-    >>> f = pa.BufferReader(data)
-    >>> f.size()
+    >>> buf = memoryview(data)
```

Comment:
Is the memoryview(..) needed?

Reply:
It needs some kind of a source ([str, Path, buffer, file-like object, …](https://docs.python.org/3/library/stdtypes.html#str)):

```python
>>> import pyarrow as pa
>>> data = b'reader data'
>>> with pa.input_stream(data) as stream:
...     stream.size()
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyarrow/io.pxi", line 2403, in pyarrow.lib.input_stream
    hasattr(source, 'closed')):
TypeError: pa.input_stream() called with instance of '<class 'bytes'>'
```
Can change the source, if preferred.